### ISSUE TITLE
HBA: Enforcing transport ssl breaks other authentication methods

### ISSUE THREAD

**Issue by fetzerms** [crate/crate#11856]
### CrateDB version

4.6.4

### CrateDB setup information

Number of nodes: 1 (just for trying out)

```
 auth.host_based.enabled: true
 auth:
   host_based:
     config:
       0:
         user: crate
         address: _local_
         method: trust
       1: 
         user: crate
         address: 10.10.10.10
         method: trust
         protocol: http
         ssl: on
       2:
         protocol: transport
         method: cert
         ssl: on
       3:
         protocol: http
         ssl: on
         method: password

<- snip ->
 # Enable encrypted communication for the HTTP endpoints:
 ssl.http.enabled: true
 
 # Enable encrypted communication for the PostgreSQL wire protocol:
 ssl.psql.enabled: true
 
 # Encrypted communication within the cluster
 ssl.transport.mode: on
<- snip ->

```

### Steps to Reproduce

1. Spin up a fresh crate installation
2. Configure a ssl keystore
3. Configure HBA and SSL as in the snipping above.
4. Try to connect from 10.10.10.10 with curl

### Expected Result

My expected result is:

- Localhost can connect on any protocol as the super user
- 10.10.10.10 can connect using https as the super user
- Other nodes can connect using transport/ssl with certificate auth
- Everything else requires https and a password

### Actual Result

To me it seems that crate thinks that the clients need to be authenticated via certificates.
This happens from localhost, 10.10.10.10 and from clients that do not have a matching host in the config.
```
$ curl -4 https://testhost1:4200/ --insecure
curl: (56) OpenSSL SSL_read: error:14094412:SSL routines:ssl3_read_bytes:sslv3 alert bad certificate, errno 0

```

If I remove the transport entry

```
       2:
         protocol: transport
         method: cert
         ssl: on
```

It seems to be work correctly. Localhost and 10.10.10.10 work directly, from somewhere else I need a password.

**Comment by BaurzhanSakhariev** [comment]
Hi, thanks for your report. 

Could you please post a result of curl with verbose? `curl -4 --verbose https://testhost1:4200/ --insecure`

**Comment by fetzerms** [comment]
I am currently on vacation and back next week. I'll send an update as soon as I can.

**Comment by fetzerms** [comment]
Ok, here are the (censored) outputs:

With the transport block:
```
*   Trying 10.10.10.1:4200...
* TCP_NODELAY set
* Connected to testhost1 (10.10.10.1) port 4200 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
* successfully set certificate verify locations:
*   CAfile: /etc/ssl/certs/ca-certificates.crt
  CApath: /etc/ssl/certs
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
* TLSv1.3 (IN), TLS handshake, Server hello (2):
* TLSv1.3 (IN), TLS handshake, Encrypted Extensions (8):
* TLSv1.3 (IN), TLS handshake, Request CERT (13):
* TLSv1.3 (IN), TLS handshake, Certificate (11):
* TLSv1.3 (IN), TLS handshake, CERT verify (15):
* TLSv1.3 (IN), TLS handshake, Finished (20):
* TLSv1.3 (OUT), TLS change cipher, Change cipher spec (1):
* TLSv1.3 (OUT), TLS handshake, Certificate (11):
* TLSv1.3 (OUT), TLS handshake, Finished (20):
* SSL connection using TLSv1.3 / TLS_AES_256_GCM_SHA384
* ALPN, server did not agree to a protocol
* Server certificate:
*  subject: C=Unknown; O=Unknown; OU=; CN=testhost1
*  start date: Oct 31 08:19:29 2021 GMT
*  expire date: Oct  7 08:19:29 2121 GMT
*  issuer: C=EU; ST=Internet; O=censored
*  SSL certificate verify result: self signed certificate in certificate chain (19), continuing anyway.
> GET / HTTP/1.1
> Host: testhost1:4200
> User-Agent: curl/7.68.0
> Accept: */*
> 
* TLSv1.3 (IN), TLS alert, bad certificate (554):
* OpenSSL SSL_read: error:14094412:SSL routines:ssl3_read_bytes:sslv3 alert bad certificate, errno 0
* Closing connection 0
curl: (56) OpenSSL SSL_read: error:14094412:SSL routines:ssl3_read_bytes:sslv3 alert bad certificate, errno 0
```

Without the transport block:
```
curl https://testhost1:4200 --insecure -4 --verbose
*   Trying 10.10.10.1:4200...
* TCP_NODELAY set
* Connected to testhost1 (10.10.10.1) port 4200 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
* successfully set certificate verify locations:
*   CAfile: /etc/ssl/certs/ca-certificates.crt
  CApath: /etc/ssl/certs
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
* TLSv1.3 (IN), TLS handshake, Server hello (2):
* TLSv1.3 (IN), TLS handshake, Encrypted Extensions (8):
* TLSv1.3 (IN), TLS handshake, Certificate (11):
* TLSv1.3 (IN), TLS handshake, CERT verify (15):
* TLSv1.3 (IN), TLS handshake, Finished (20):
* TLSv1.3 (OUT), TLS change cipher, Change cipher spec (1):
* TLSv1.3 (OUT), TLS handshake, Finished (20):
* SSL connection using TLSv1.3 / TLS_AES_256_GCM_SHA384
* ALPN, server did not agree to a protocol
* Server certificate:
*  subject: C=Unknown; O=Unknown; OU=; CN=testhost1
*  start date: Oct 31 08:19:29 2021 GMT
*  expire date: Oct  7 08:19:29 2121 GMT
*  issuer: C=EU; ST=Internet; O=censored
*  SSL certificate verify result: self signed certificate in certificate chain (19), continuing anyway.
> GET / HTTP/1.1
> Host: testhost1:4200
> User-Agent: curl/7.68.0
> Accept: */*
> 
* TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
* Mark bundle as not supporting multiuse
< HTTP/1.1 503 Service Unavailable
< content-type: application/json
< content-length: 319
< 
{
  "ok" : false,
  "status" : 200,
  "name" : "testhost1",
  "cluster_name" : "crate",
  "version" : {
    "number" : "4.6.4",
    "build_hash" : "44069f68c7242d3ae55ef5ce4c9f736486128bf3",
    "build_timestamp" : "2021-09-30T13:28:49Z",
    "build_snapshot" : false,
    "lucene_version" : "8.8.2"
  }
}
* Connection #0 to host testhost1 left intact
```

To me, it does not seem very informative, but it seems that with the transport block client cert auth is tried.

**Comment by BaurzhanSakhariev** [comment]
>TLSv1.3 (IN), TLS handshake, Request CERT (13):
TLSv1.3 (OUT), TLS handshake, Certificate (11):

Ok, that's a bug indeed. Thanks for the reporting, will fix it soon.

**Comment by fetzerms** [comment]
Hi. Thanks for confirming the bug. You are right, the 503 error is something that I didn't care about (yet :-))

**Comment by BaurzhanSakhariev** [comment]
Fixed, will be available in the next hot fix release.

In a meantime possible workaround -  when I started a node with the same config on my machine and did a `curl` from the same machine it worked - because I [added self signed cert into the store](https://crate.io/docs/crate/reference/en/4.6/admin/ssl.html#id17) and crate node could recognise it since it used the same store.

Crate node (server) has client auth which can be no, optional and required - and you faced optional because of the bug. Normally its a big deal since certificate can be not provided at all(because it's optional) but **if it's provided it should be valid/trusted** and curl reacted to `TLSv1.3 (IN), TLS handshake, Request CERT (13): `  by providing a cert which was rejected by the node. 

`--insecure` option is about ignoring incoming server's cert validation but not the opposite so it didn't help
